From b8e6c4a6bf5cfd1064a0d0acdb61e40b7bdf10c9 Mon Sep 17 00:00:00 2001
From: Dmitry Perchanov <dmitry.perchanov@intel.com>
Date: Thu, 27 Oct 2022 18:52:51 +0300
Subject: [PATCH] fix the retval in [accel|gyro]_3d_capture_sample Module
 version realsense2-dkms to track changes Authored by: Yu MENG
 <yu1.meng@intel.com>

Signed-off-by: Dmitry Perchanov <dmitry.perchanov@intel.com>
---
 5.15.0/drivers/iio/accel/hid-sensor-accel-3d.c | 2 ++
 5.15.0/drivers/iio/gyro/hid-sensor-gyro-3d.c   | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/5.15.0/drivers/iio/accel/hid-sensor-accel-3d.c b/5.15.0/drivers/iio/accel/hid-sensor-accel-3d.c
index a2def6f..5df6c69 100644
--- a/5.15.0/drivers/iio/accel/hid-sensor-accel-3d.c
+++ b/5.15.0/drivers/iio/accel/hid-sensor-accel-3d.c
@@ -280,6 +280,7 @@ static int accel_3d_capture_sample(struct hid_sensor_hub_device *hsdev,
 			hid_sensor_convert_timestamp(
 					&accel_state->common_attributes,
 					*(int64_t *)raw_data);
+		ret = 0;
 	break;
 	default:
 		break;
@@ -461,3 +462,4 @@ MODULE_DESCRIPTION("HID Sensor Accel 3D");
 MODULE_AUTHOR("Srinivas Pandruvada <srinivas.pandruvada@intel.com>");
 MODULE_LICENSE("GPL");
 MODULE_IMPORT_NS(IIO_HID);
+MODULE_VERSION("realsense2-dkms");
diff --git a/5.15.0/drivers/iio/gyro/hid-sensor-gyro-3d.c b/5.15.0/drivers/iio/gyro/hid-sensor-gyro-3d.c
index 8f0ad02..ba5a536 100644
--- a/5.15.0/drivers/iio/gyro/hid-sensor-gyro-3d.c
+++ b/5.15.0/drivers/iio/gyro/hid-sensor-gyro-3d.c
@@ -231,6 +231,7 @@ static int gyro_3d_capture_sample(struct hid_sensor_hub_device *hsdev,
 		gyro_state->timestamp =
 			hid_sensor_convert_timestamp(&gyro_state->common_attributes,
 						     *(s64 *)raw_data);
+		ret = 0;
 	break;
 	default:
 		break;
@@ -395,3 +396,4 @@ MODULE_DESCRIPTION("HID Sensor Gyroscope 3D");
 MODULE_AUTHOR("Srinivas Pandruvada <srinivas.pandruvada@intel.com>");
 MODULE_LICENSE("GPL");
 MODULE_IMPORT_NS(IIO_HID);
+MODULE_VERSION("realsense2-dkms");
-- 
2.37.1

